tools/profiles/debug.json

```diff
@@ -1,15 +1,15 @@
 {
     "GCC_ARM": {
-        "common": ["-c", "-Wall", "-Wextra",
+        "common": ["-Wall", "-Wextra",
                    "-Wno-unused-parameter", "-Wno-missing-field-initializers",
                    "-fmessage-length=0", "-fno-exceptions",
                    "-ffunction-sections", "-fdata-sections", "-funsigned-char",
                    "-MMD",
-                   "-fomit-frame-pointer", "-Og", "-g3", "-DMBED_DEBUG",
+                   "-fomit-frame-pointer", "-Og", "-DMBED_DEBUG",
                    "-DMBED_TRAP_ERRORS_ENABLED=1"],
-        "asm": ["-x", "assembler-with-cpp"],
-        "c": ["-std=gnu11"],
-        "cxx": ["-std=gnu++14", "-fno-rtti", "-Wvla"],
+        "asm": ["-c", "-g3", "-x", "assembler-with-cpp"],
+        "c": ["-c", "-g3", "-std=gnu11"],
+        "cxx": ["-c", "-g3", "-std=gnu++14", "-fno-rtti", "-Wvla"],
```

Comment:
These changes do not appear to be part of the work to add a profile extension. However, they look ok.

Reply:
Thanks @mark-edgeworth; I left [this commit](https://github.com/ARMmbed/mbed-os/pull/11856/commits/fb40672e4de083d04a5cea49919387a411622099) to handle the `-flto` extension, where *it is recommended that you compile all the files participating in the same link with the same options and also specify those options at link time*.